### PR TITLE
Floating label for huge input

### PR DESCRIPTION
**Pull request by guern1kn**
## What does this PR do?
Adds floating label functionality to huge input

### Associated Issue 
#1500 

### Documentation
 - [x] Update Spark Docs Vanilla
 - [x] Update Component Sass Var/Class Modifier table

### Code
 - [x] Build Component in Spark Vanilla (Sass, HTML, JS)
 - [x] Unit Testing in Spark Vanilla with `npm run test` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### Design Review
 - [x] Design reviewed and approved

### CODE REVIEW

**Comment by RVMendoza** [comment]
@afebbraro fixing the direct selector feedback in the next commit

**Comment by RVMendoza** [comment]
@afebbraro I agree with you about keeping our downward spacing flow.

I believe this code was following what already existed right above it. Probably why that choice was made.
image

I put a ToDo on the code when we start working on icon on huge input later (we lessened scope so we can move forward)

File: packages/spark/settings/_settings.scss [hunk 1]
```diff
@@ -443,15 +443,17 @@ $sprk-text-input-font-weight: 300 !default;
 $sprk-text-input-height: 48px !default;
```
Comment (13protons) [inline packages/spark/settings/_settings.scss]: @guern1kn I don't think you should change anything right now, but I think we need to discuss our variable naming and inheritance strategy as a team. There's so many things here that should probably just inherit form other vars and not just be their stand values.

Comment (guern1kn) [reply packages/spark/settings/_settings.scss]: Sounds like a good meta topic!

Comment (yodasw16) [reply packages/spark/settings/_settings.scss]: This is probably part of a bigger discussion about Design Tokens
https://github.com/sparkdesignsystem/spark-design-system/issues/1120

**Comment by bobwise** [comment]
Observed VoiceOver behavior:
image

We may want to take another look at how the structure is being interpreted for screen readers. Huge Input behavior is a little bit different from other inputs.

Expected screen reader behavior:
1. First focus and describe the label - "Text Input Label"
2. Then focus and describe the input
 2a. read the placeholder - "Placeholder"
 2b. then read the label - "Text Input Label"
 2c. then describe the state - "edit text"

File: packages/spark/base/inputs/_svg-icon-input.scss [hunk 2]
```diff
@@ -4,7 +4,15 @@
 
 .sprk-b-TextInputIconContainer .sprk-c-Icon {
   margin-left: $sprk-input-icon-offset-x;
+  // TODO: We should be using margin bottom, not margin top
```
Comment (RCopeland) [inline packages/spark/base/inputs/_svg-icon-input.scss]: do we need an issue for this?

Comment (guern1kn) [reply packages/spark/base/inputs/_svg-icon-input.scss]: Probably? I'll send a note to steph.

**Pull request merged**